homeassistant/components/flux.py

```diff
@@ -0,0 +1,228 @@
+"""
+Flux for Home-Assistant.
+
+The primary functions in this component were taken from
+https://github.com/KpaBap/hue-flux/
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/flux/
+"""
+from datetime import timedelta
+import logging
+import math
+import voluptuous as vol
+
+from homeassistant.helpers.event import track_time_change
+from homeassistant.components.light import is_on, turn_on, turn_off
+from homeassistant.components.sun import next_setting, next_rising
+import homeassistant.helpers.config_validation as cv
+import homeassistant.util.dt as dt_util
+
+DEPENDENCIES = ['sun', 'light']
+DOMAIN = "flux"
+SUN = "sun.sun"
+_LOGGER = logging.getLogger(__name__)
+
+CONF_LIGHTS = 'lights'
+CONF_WAKETIME = 'waketime'
+CONF_BEDTIME = 'bedtime'
+CONF_TURNOFF = 'turn_off'
+CONF_AUTO = 'auto'
+CONF_DAY_CT = 'day_colortemp'
+CONF_SUNSET_CT = 'sunset_colortemp'
+CONF_BEDTIME_CT = 'bedtime_colortemp'
+CONF_BRIGHTNESS = 'brightness'
+
+PLATFORM_SCHEMA = vol.Schema({
+    vol.Required(CONF_LIGHTS): [cv.string],
+    vol.Optional(CONF_WAKETIME): vol.Coerce(str),
```

Comment:
At top of file:
```python
from homeassistant.helpers import config_validation as cv
```

After that, you can use the `cv.time` validator for timestamps.